slackdump/slackhistory.py

```diff
@@ -0,0 +1,116 @@
+import time
+import hashlib
+
+from slacker import Slacker
+from pymongo import MongoClient, ASCENDING, DESCENDING
+
+
+class SlackHistory(object):
+
+    def __init__(self, auth_token = 0, connect_time = time.time()):
+        self.client = MongoClient()
+        self.db = self.client.slackdb
+        self.connect_time = connect_time
+        self.auth_token = auth_token
+        self.slack = Slacker(self.auth_token)
+        self._collection = self.db.slackdb
+
+    def start(self):
+        self.get_public_channels_history()
+        self.get_private_channels_history()
+        self.get_direct_channels_history()
+
+    def get_key(self, message):
+        return hashlib.sha1((message['channel']
+                             + message['ts']).encode('utf8')).hexdigest()
+
+    def insert_into_db(self, message):
+        self.db.slackdb.insert_one(message)
+        print("Inserted")
+
+    def get_history(self, slack_object, channel_id, end_time, start_time = 0):
+        messages = []
+        last_time_stamp = end_time
+        oldest_time_stamp = start_time
+
+        while True:
+            response = slack_object.history(channel_id,
+                                            last_time_stamp, oldest_time_stamp,
+                                            count = 100).body
+
+            messages.extend(response['messages'])
+
```

Comment:
when we start a loop there is no need to separate it with above code, but when it ends it is required.